crates/sui-framework/src/natives/object_runtime/mod.rs

```diff
@@ -113,6 +113,10 @@ impl<'a> ObjectRuntime<'a> {
     }
 
     pub fn new_id(&mut self, id: ObjectID) {
+        // remove if from deleted_ids for the case in dynamic fields where the
```

Comment:
```suggestion
        // remove from deleted_ids for the case in dynamic fields where the
```

Reply:
Is there a representation of `state` that combines `new_ids`, `deleted_ids`, etc into a single map with an enum value type where updating that map gives you this property for free?

Reply:
Also, this comment paints a quite general picture of object IDs being deleted and then re-added.  Is it worth clarifying that this only happens in the narrow circumstance of a `Field` type being removed and then re-added, because otherwise it might give the impression that this is admissible for IDs in general.

Reply:
> Is there a representation of state that combines new_ids, deleted_ids, etc into a single map with an enum value type where updating that map gives you this property for free?

Possibly... but I think it would be outside of the scope of the PR. I can't think of something general that gives us all the properties we want by construction. Even switching it to being a mapping `ID => (New | Deleted)` doesn't quite work for the case of deleting something where `ID => New`, we don't want `ID => Deleted`, we want `ID => _`.

> Also, this comment paints a quite general picture of object IDs being deleted and then re-added. Is it worth clarifying that this only happens in the narrow circumstance of a Field type being removed and then re-added, because otherwise it might give the impression that this is admissible for IDs in general.

Will do